tsdb/chunks/head_chunks.go

```diff
@@ -907,14 +923,27 @@ func (cdm *ChunkDiskMapper) DeleteCorrupted(originalErr error) error {
 	// Delete all the head chunk files following the corrupt head chunk file.
 	segs := []int{}
 	cdm.readPathMtx.RLock()
+	lastSeq := 0
 	for seg := range cdm.mmappedChunkFiles {
 		if seg >= cerr.FileIndex {
 			segs = append(segs, seg)
+		} else if seg > lastSeq {
+			lastSeq = seg
 		}
 	}
 	cdm.readPathMtx.RUnlock()
 
-	return cdm.deleteFiles(segs)
+	pendingDeletes, err := cdm.deleteFiles(segs)
+	cdm.evtlPosMtx.Lock()
+	if err == nil {
+		cdm.evtlPos.setSeq(uint64(lastSeq))
```

Comment:
Isn't it possible that `DeleteCorrupted` will delete all files in the fs? In that case this would have to be set to `0`.

I'm starting to wonder if it wouldn't be easier to just add a method to `ChunkDiskMapper` which looks at the fs to determine the highest present chunk and then returns its sequence, similar like what happens in `openMMapFiles()`. This could then be used here and also in `openMMapFiles()`, not in `Truncate()` though because at the point where `Truncate()` is called there might chunks in the queue.

What do you think about this?

Reply:
`lastSeq` starts at 0, and will be never set if all files are deleted. So it will be 0, right?

And about checking fs to determine the last sequence: I thought about that, but error handling becomes more complect. What if traversing the directory gives and error? What do we do then?

Reply:
We are doing that
```go
		} else if seg > lastSeq {
			lastSeq = seg
		}
```

Reply:
🤦‍♂️ sorry, you're right again